ti/instrumentation/utils.go

```diff
@@ -264,6 +266,44 @@ func getChangedFiles(ctx context.Context, workspace string, log *logrus.Logger,
 	return res, nil
 }
 
+// addBazelFilesToChangedFiles takes a list of changed files and removes bazel build files and adds files in that package to the changed file list
+func addBazelFilesToChangedFiles(ctx context.Context, workspace string, log *logrus.Logger, res []ti.File) ([]ti.File, error) {
+	//resCopy := copy(resCopy, res)
+	changedRes := []ti.File{}
+	//copy(copiedRes, res)
+
+	for _, file := range res {
+		log.Infoln("A, currently processing file : ", file)
+		if strings.HasSuffix(string(file.Name), "BUILD.bazel") {
+			fileName := "//" + strings.Replace(file.Name, "/BUILD.bazel", "", -1)
+			c := fmt.Sprintf("cd %s; %s query 'kind(\"java_library\", %s:tests)' --output=build "+
```

Comment:
Can you add some comments explaining what this long command does? Also, you are looking at `java_library` only. Is that the only target/rule to look for?

Reply:
Modified to include rules beginning with "java", so java_library and java_binary